File: sdks/go/pkg/models/operations/getfilesidmetadata.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetFilesIDMetadataPathParams struct {
	FileID string `pathParam:"style=simple,explode=false,name=file_id"`
}

type GetFilesIDMetadataRequest struct {
	Retries    *utils.RetryConfig
	PathParams GetFilesIDMetadataPathParams
}

type GetFilesIDMetadataResponse struct {
	ClientError *shared.ClientError
	ContentType string
	Metadatas   *shared.Metadatas
	StatusCode  int64
}

File: sdks/ts/src/sdk/models/operations/getfilesidmetadata.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetFilesIdMetadataPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=file_id" })
  fileId: string;
}


export class GetFilesIdMetadataRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetFilesIdMetadataPathParams;
}


export class GetFilesIdMetadataResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  metadatas?: shared.Metadatas;

  @SpeakeasyMetadata()
  statusCode: number;
}
